js/field.js

```diff
@@ -0,0 +1,324 @@
+/**
+ * Created by Ruslan on 29-Mar-16.
+ */
+
+class Field {
+  constructor(options) {
+    this._el = options.element;
+    this._cards = options.cards;
+    this._currrentZindex = 50;
+
+    this._cardDeckTemplate = document.getElementById('cards-deck-template').innerHTML;
+
+    this._deck = new CardHolder({
+      element: this._el.querySelector('[data-component="deck"]')
+    });
+
+    this._openDeck = new CardHolder({
+      element: this._el.querySelector('[data-component="open-deck"]')
+    });
+
+    this._dragManager = new DragManager({
+      element: this._el.querySelector('[data-component="drag-manager"]')
+    });
+
+    this._initializePiles();
+    this._initializeHomes();
+    this._addAllCardsToDeck();
+    this._addStartCardsToPiles();
+
+    if (this._deck.topCardElement) {
+      this._deck.topCardElement.onclick = this._onDeckTopCardClick.bind(this);
+    }
+
+    this._deck.on('holderClick', this._onDeckClick.bind(this));
+    this._dragManager.on('moveElement', this._onDragManagerMoveElement.bind(this));
+  }
+
+  _onDeckClick(event) {
+    this._moveCardsFromOpenDeckToDeck();
+  }
+
+  _onDeckTopCardClick(event) {
+    this._moveCardFromDeckToOpenDeck();
+  }
+
+  _onDragManagerMoveElement(event) {
+    let toId = event.detail.destinationElementId;
+    let cardId = event.detail.movingElementId;
+
+    switch (toId) {
+      case 'home-0' :
+        this._addCardToHome(0, cardId);
+        break;
+      case 'home-1' :
+        this._addCardToHome(1, cardId);
+        break;
+      case 'home-2' :
+        this._addCardToHome(2, cardId);
+        break;
+      case 'home-3' :
+        this._addCardToHome(3, cardId);
+        break;
+      default : // card to card or card to empty pile
```

Comment:
The comments are in different languages. It is better to have only one through all files.